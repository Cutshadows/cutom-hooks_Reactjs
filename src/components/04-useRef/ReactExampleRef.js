import React, {  useState } from 'react';
import '../02-useEffect/effect.scss';
import MultipleCustomHooks from '../03-examples/MultipleCustomHooks';

const ReactExampleRef = () => {

    const [show, setShow] = useState(false);

    return ( 
        <div>
            <h1>ReactExampleRef</h1>
            <hr/>

            {show&&<MultipleCustomHooks />}

            <button
            className="btn btn-primary mt-5"
            onClick={()=>{
                setShow(!show)
            }}
            >
                Show/Hide
            </button>
        </div>
     );
}
 
export default ReactExampleRef;