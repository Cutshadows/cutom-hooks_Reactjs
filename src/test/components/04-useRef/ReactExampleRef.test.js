import {useState} from 'react';
import {act, renderHook} from '@testing-library/react-hooks';
import { shallow } from 'enzyme';
import ReactExampleRef from '../../../components/04-useRef/ReactExampleRef';


describe('Pruebas en <ReactExampleRef></ReactExampleRef>', () => {
    const wrapper=shallow(<ReactExampleRef/>);

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
    })
    
    test('debe mostrarse el componente <MultipleCustomHooks></MultipleCustomHooks>', () => {
        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);

        const {result}=renderHook(()=>useState(false));
        const [,setShow]=result.current;
        act(()=>{
           setShow(true)
        })
        const[show]=result.current;
        expect(show).toBe(true);


    })
    
    
})
