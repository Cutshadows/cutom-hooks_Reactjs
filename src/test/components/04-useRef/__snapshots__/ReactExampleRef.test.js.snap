// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ReactExampleRef></ReactExampleRef> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    ReactExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
